backend/api/views/main.py

```diff
@@ -164,8 +164,150 @@ def update_documents(docClass):
     return create_response(status=200, message="success")
 
 
-# function that is called when you visit /field_partner
+# function that is called when you visit /field_partner, gets all the FPs
 @main.route("/field_partner", methods=["GET"])
 def get_field_partner():
     field_partner = FieldPartner.query.all()
     return create_response(data={"field_partner": serialize_list(field_partner)})
+
+
+# function that is called when you visit /field_partner/get/id/<id> that gets a field partner by id
+@main.route("/field_partner/id/<id>", methods=["GET"])
+def get_fp_by_id(id):
+    field_partner_by_id = FieldPartner.query.get(id)
+    return create_response(data={"field_partner": field_partner_by_id.to_dict()})
```

Comment:
Good comments for the functions, but I would like them to be inside the function (either as a docstring or as the way you have them currently)